### PR TITLE
MC-38910 Add compatibility with composer 2

### PR DESCRIPTION
**Pull request by jcuerdo**

### CODE REVIEW

**Review by loginesta — APPROVED** [review]

**Review by fascinosum — CHANGES_REQUESTED** [review]

File: composer.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "repositories": [
         {
             "type": "vcs",
-            "url": "git@github.com:Flyingmana/phpcs.git"
+            "url": "git@github.com:firegento/phpcs.git"
```
[the remark this answers is not in the document]
Comment (jcuerdo) [reply composer.json:12]: Flyingmana/phpcs does not exists anymore. Moved to firegento/phpcs

File: composer.json [hunk 2]
```diff
@@ -39,16 +39,16 @@
...
-        "composer-plugin-api": "^1.0"
+        "composer-plugin-api": "^1.0|^2.0"
     },
     "require-dev":{
         "phpunit/phpunit":"*",
         "phpunit/phpunit-mock-objects": "dev-master",
         "squizlabs/php_codesniffer": "1.4.7",
-        "firegento/phpcs": "dev-patch-1",
-        "composer/composer":"*@dev",
+        "firegento/phpcs": "dev-develop",
```
[the remark this answers is not in the document]
Comment (jcuerdo) [reply composer.json:48]: You right ¡¡ Let's use ^1.0.1

File: composer.json [hunk 3]
```diff
@@ -39,16 +39,16 @@
...
+        "composer/composer":"<=1.10.6|| ^2.0",
```
[the remark this answers is not in the document]
Comment (jcuerdo) [reply composer.json:49]: @Rrego6 you think we could use "^1.0" instead ?

Will be like "^1.0|| ^2.0"

[the remark this answers is not in the document]
Comment (jcuerdo) [reply composer.json]: "^2.0" is enought

**Pull request closed without merging**